text/0510-dynamodb-global-table.md

```diff
@@ -0,0 +1,1742 @@
+* `tableClass` --> Gets copied over from table level props if defined.
+* `tags` --> Gets copied over from table level props if defined.
+* `readWithAutoScaling` --> Gets copied over from table level props if defined.
+* `readCapactity` --> Gets copied over from table level props if defined.
+* `encryptionKey` --> This needs to be defined per replica
+* `kinesisStream` --> This needs to be defined per replica
+* `gsiProps`
+  * `indexName` --> Needs to be specified
+  * `contributorInsightsEnabled` --> Gets copied over from table level props or replica level props if defined.
+
+You can also add a replica using `addReplica` method.
+
+```typescript
+  globalTable.addReplica({
+    region: 'us-east-1',
+  });
+```
+
+__NOTE:__
+
+* A replica needs to be deployed to the region where stack is deployed to. We do this automatically if
+user has not defined the region in replica.
+
+#### Grants
+
+You can grant access to certain operation on the table with the help of `grant` functions.
+
+```typescript
+  const globalTable = new GlobalTable(stack, 'FooTable', {
+    partitionKey: { name: 'Foo', type: AttributeType.STRING },
+    replicas: [
+      {
+        region: 'us-west-2',
+      },
+    ],
+  });
+
+  const lambdaRole = new iam.Role(stack, 'Role', {
+    assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
+  });
+
+  globalTable.grantReadData(lambdaRole, ['us-west-2']);
+```
+
+#### Metrics
+
+You can access metrics emitted by global table by using `metric` functions.
+
+```typescript
+  const globalTable = new GlobalTable(stack, 'FooTable', {
+    partitionKey: { name: 'Foo', type: AttributeType.STRING },
+    replicas: [
+      {
+        region: 'us-west-2',
+      },
+    ],
+  });
+
+  const metric = globalTable.metricConsumedWriteCapacityUnits({
+    region: 'us-west-2',
+  });
+```
+
+#### Importing
+
+You can import an existing global table in your stack by using `from` functions. You would need to specify
+ either table name, table arn or table attributes to import it.
+
+```typescript
+GlobalTable.fromTableName(stack, 'FooTableId', 'FooTable');
+```
+
+### NOTE
+
+* Be careful while naming your global table. If there is a name conflict in a region with a DynamoDB table,
+that could lead to removal of that table with the global table replica.
+* We only support version [2019.11.21](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/globaltables.V2.html)
+of the global table since CloudFormation just has support for the same.
+
+---
+
+Ticking the box below indicates that the public API of this RFC has been
+signed-off by the API bar raiser (the `api-approved` label was applied to the
+RFC pull request):
+
+```
+[ ] Signed-off by API Bar Raiser @xxxxx
+```
+
+## Public FAQ
+
+### What are we launching today?
+
+We are launching L2 support for DynamoDB Global Table feature.
+
+### Why should I use this feature?
+
+You should use this feature if you would like to provision a DynamoDB table that has the capability
```

Comment:
You would also need to use this feature if:

- You want to import tables using *CloudFormation import*
- You want to use drift detection
- You want to successfully create replicated GSIs for an autoscaled table (our current custom resource doesn't do that properly)